test/unit_test/rag/llm/test_embedding_drop_params.py

```
@@ -0,0 +1,182 @@
+    def __init__(self, vectors):
+        self.data = [SimpleNamespace(embedding=list(v)) for v in vectors]
+        self.usage = SimpleNamespace(total_tokens=0)
+
+
+def _make_openai_embed(cls=OpenAIEmbed):
+    """Construct an embedding client and capture the kwargs of every
+    ``embeddings.create`` call into ``embed.client.embeddings.create.call_args_list``.
+    """
+    embed = cls("key", "text-embedding-3-small", base_url="https://example.invalid/v1")
+    embed.client = MagicMock()
+    embed.client.embeddings.create = MagicMock(side_effect=lambda input, model, **kwargs: _FakeResp([[float(i)] for i in range(len(input))]))
```

Ruff (0.16.1)

[error] Lambda argument `input` is shadowing a Python builtin

(A006)